image-setup: run iotune for instances with no pre-configured io

benipeled opened the pull request:
We need to make sure that instances with no pre-configured IO, will run
`iotune` during the boot (as part of scylla-image-setup service),

@tarzanek the `run_iotune` function added on scylla_cloud_io_setup is a copy-paste of the original function you added [1] but without the cpuset/smp flag condition - not sure it's required - please take a look,

I verified this change on r5d.xlarge and i4i.xlarge and it works as expected but the fallback skipped for r5.xlarge since it's not part of the supported instance list [1] - what was the original behavior? did this work for diskless instances like r5.xlarge as well?

[0] https://github.com/scylladb/scylla-enterprise/blob/af3da2d7ac8b83e1710c418b269c3c85bd1fdbe4/dist/common/scripts/scylla_io_setup#L113-L165

[1] 'i2', 'i3', 'i3en', 'c5d', 'm5d', 'm5ad', 'r5d', 'z1d', 'c6gd', 'm6gd', 'r6gd', 'x2gd', 'im4gn', 'is4gen', 'i4i'

Fix #376

tarzanek reviewed (approved):
lgtm
btw. this (back)porting work looks error prone, are we duplicating code?

benipeled commented:
> lgtm btw. this (back)porting work looks error prone, are we duplicating code?

indeed, I thought it was removed from master but it was my fault (tx @yaronkaikov)

I remove the duplication code, the service will run scylla_io_setup in case of no-preconfigure io

yaronkaikov reviewed (approved):
LGTM

tarzanek reviewed (approved):
lgtm

benipeled commented:
Verified with r5d.xlarge
```
scyllaadm@ip-10-0-1-251:~$ systemctl status scylla-image-setup
● scylla-image-setup.service - Scylla Cloud Image Setup service
     Loaded: loaded (/lib/systemd/system/scylla-image-setup.service; enabled; vendor preset: enabled)
     Active: active (exited) since Tue 2022-07-26 09:41:53 UTC; 49s ago
    Process: 458 ExecStart=/opt/scylladb/scylla-machine-image/scylla_image_setup (code=exited, status=0/SUCCESS)
   Main PID: 458 (code=exited, status=0/SUCCESS)

Jul 26 09:39:38 ip-10-0-1-251 scylla_image_setup[708]:          =                       sectsz=512   sunit=0 blks, lazy-count=1
Jul 26 09:39:38 ip-10-0-1-251 scylla_image_setup[708]: realtime =none                   extsz=4096   blocks=0, rtextents=0
Jul 26 09:39:38 ip-10-0-1-251 scylla_image_setup[766]: Created symlink /etc/systemd/system/multi-user.target.wants/var-lib-scylla.mount → /etc/systemd/system/var-lib-scylla.mount.
Jul 26 09:39:38 ip-10-0-1-251 scylla_image_setup[799]: update-initramfs: Generating /boot/initrd.img-5.15.0-1015-aws
Jul 26 09:39:48 ip-10-0-1-251 scylla_image_setup[690]: Creating XFS volume for scylla using 1 disk(s): /dev/nvme3n1
Jul 26 09:39:48 ip-10-0-1-251 scylla_image_setup[681]: Devices: ['/dev/nvme3n1']
Jul 26 09:39:49 ip-10-0-1-251 scylla_image_setup[458]: WARNING:root:This is a supported instance but with no pre-configured io, scylla_io_setup will be run
Jul 26 09:41:52 ip-10-0-1-251 scylla_image_setup[5567]: Created symlink /etc/systemd/system/multi-user.target.wants/var-lib-systemd-coredump.mount → /etc/systemd/system/var-lib-systemd-coredump.mount.
Jul 26 09:41:52 ip-10-0-1-251 scylla_image_setup[5606]: Removed /etc/systemd/system/timers.target.wants/fstrim.timer.
Jul 26 09:41:53 ip-10-0-1-251 systemd[1]: Finished Scylla Cloud Image Setup service.
```

and with i4i.xlarge
```
scyllaadm@ip-10-0-1-209:~$ systemctl status scylla-image-setup
● scylla-image-setup.service - Scylla Cloud Image Setup service
     Loaded: loaded (/lib/systemd/system/scylla-image-setup.service; enabled; vendor preset: enabled)
     Active: active (exited) since Tue 2022-07-26 09:46:34 UTC; 2min 10s ago
    Process: 456 ExecStart=/opt/scylladb/scylla-machine-image/scylla_image_setup (code=exited, status=0/SUCCESS)
   Main PID: 456 (code=exited, status=0/SUCCESS)

Jul 26 09:46:26 ip-10-0-1-209 scylla_image_setup[716]: log      =internal log           bsize=1024   blocks=446796, version=2
Jul 26 09:46:26 ip-10-0-1-209 scylla_image_setup[716]:          =                       sectsz=512   sunit=0 blks, lazy-count=1
Jul 26 09:46:26 ip-10-0-1-209 scylla_image_setup[716]: realtime =none                   extsz=4096   blocks=0, rtextents=0
Jul 26 09:46:26 ip-10-0-1-209 scylla_image_setup[769]: Created symlink /etc/systemd/system/multi-user.target.wants/var-lib-scylla.mount → /etc/systemd/system/var-lib-scylla.mount.
Jul 26 09:46:26 ip-10-0-1-209 scylla_image_setup[802]: update-initramfs: Generating /boot/initrd.img-5.15.0-1015-aws
Jul 26 09:46:33 ip-10-0-1-209 scylla_image_setup[688]: Creating XFS volume for scylla using 1 disk(s): /dev/nvme3n1
Jul 26 09:46:33 ip-10-0-1-209 scylla_image_setup[679]: Devices: ['/dev/nvme3n1']
Jul 26 09:46:34 ip-10-0-1-209 scylla_image_setup[5463]: Created symlink /etc/systemd/system/multi-user.target.wants/var-lib-systemd-coredump.mount → /etc/systemd/system/var-lib-systemd-coredump.mount.
Jul 26 09:46:34 ip-10-0-1-209 scylla_image_setup[5502]: Removed /etc/systemd/system/timers.target.wants/fstrim.timer.
Jul 26 09:46:34 ip-10-0-1-209 systemd[1]: Finished Scylla Cloud Image Setup service.
```

As mentioned above, non-disk instances like r5.xlarge have no automatic-tuning but I see no reason to add it
@syuu1228 any reason why not to remove the following check and allow automatic tuning to all instances?:
https://github.com/scylladb/scylla-machine-image/blob/3f633aeaeb00750ae1e244f9506304b6b6620cd6/common/scylla_image_setup#L22-L23

Full verification job https://jenkins.scylladb.com/job/benipeled_test/250/ (copy of next-machine-image job)
